src/Nfts/NftsNavigator.tsx

```diff
@@ -3,63 +3,47 @@ import React, {useState} from 'react'
 import {defineMessages, useIntl} from 'react-intl'
 import {StyleSheet, TouchableOpacity, View} from 'react-native'
 
-import {Icon, SearchBar} from '../components'
+import {Icon} from '../components'
 import {NftRoutes} from '../navigation'
+import {SearchHeader, SearchProvider} from '../Search'
 import {Nfts} from './Nfts'
 
 const Stack = createStackNavigator<NftRoutes>()
 
 export const NftsNavigator = () => {
   const strings = useStrings()
-  const [showSearch, setShowSearch] = useState(false)
-  const [search, setSearch] = useState('')
-
-  const handleClearPress = () => {
-    setSearch('')
-  }
-
-  const handleBackPress = () => {
-    setShowSearch(false)
-    setSearch('')
-  }
+  const [searchVisible, setSearchVisible] = useState(false)
 
   return (
-    <Stack.Navigator
-      screenOptions={{
-        headerTitleContainerStyle: {width: '100%', alignItems: 'center'},
-        cardStyle: {backgroundColor: '#fff'},
-      }}
-      initialRouteName="nfts"
-    >
-      <Stack.Screen
-        name="nfts"
-        options={{
-          title: strings.title,
-          headerTitleAlign: 'center',
-          header: showSearch
-            ? () => (
-                <SearchBar
-                  placeholder={strings.search}
-                  onChangeText={(search) => setSearch(search)}
-                  value={search}
-                  onClearPress={handleClearPress}
-                  onBackPress={handleBackPress}
-                />
-              )
-            : undefined,
-          headerLeft: () => <View style={styles.iconPlaceholder} />,
-          headerRight: () => (
-            <TouchableOpacity onPress={() => setShowSearch(true)} style={styles.center}>
-              <Icon.Magnify size={26} />
-            </TouchableOpacity>
-          ),
-          headerLeftContainerStyle: {paddingLeft: 16},
-          headerRightContainerStyle: {paddingRight: 16},
+    <SearchProvider>
+      <Stack.Navigator
+        screenOptions={{
+          headerTitleContainerStyle: {width: '100%', alignItems: 'center'},
+          cardStyle: {backgroundColor: '#fff'},
         }}
+        initialRouteName="nfts"
       >
-        {(props) => <Nfts {...props} search={search} />}
-      </Stack.Screen>
-    </Stack.Navigator>
+        <Stack.Screen
+          name="nfts"
+          options={{
+            title: strings.title,
+            headerTitleAlign: 'center',
+            header: searchVisible
```

Comment:
I'd wrap it in a `ToogleabeSearchBar`, but by changing the state within the navigator, all stack screens will rerender, by nesting it inside this other component only it's children will be affected.